API-Samples/run_all_samples.py

```diff
@@ -11,13 +11,20 @@
     else:
         sys.exit("Cannot find samples in Debug or Release, have they been built?")
 else:
-    directory = os.cwd()
-    
-for file in os.listdir(directory):
-    if file.endswith(".exe"):
-        sample = os.path.join(directory, file)
-        if "VK_LAYER_PATH" not in os.environ:
-            if file == "enable_validation_with_callback.exe" or file == "validation_cache.exe":
-                continue
-        print("Running: ", sample)
-        subprocess.check_call(sample)
+    directory = os.getcwd()
+
+samples = []
+samplesdir = os.path.join("..", "..", "API-Samples")
+for root, dir, files in os.walk(samplesdir):
+    for file in files:
+        if file.endswith(".cpp") and "utils" not in root and "android" not in root:
+            samples.append(os.path.splitext(file)[0])
+
+samples.sort()
+for sample in samples:
+    executable = os.path.join(directory, sample)
+    if "VK_LAYER_PATH" not in os.environ:
+        if sample == "enable_validation_with_callback" or sample == "validation_cache":
+            continue
+        print("Running: " + sample)
+        subprocess.check_call(executable)
```

Comment:
The executable here was derived from the `.cpp` files, and could feasibly not exist as an executable (the Linux counterpart to the Windows "have they been built?" condition, though this could also happen on Windows if the directory exists but some or all executables are missing).

`check_call()` will certainly raise an error if an executable is missing; that might be sufficient.  Consider, though:

```
if os.path.isfile(executable):
    print("Running: " + sample)
    subprocess.check_call(executable)
else:
    print("Skipping {} because the sample doesn't seem to be built".format(sample))
```